internal/controller/linodemachine_controller.go

```diff
@@ -513,6 +513,14 @@ func (r *LinodeMachineReconciler) reconcileUpdate(ctx context.Context, logger lo
 		machineScope.LinodeMachine.Status.Ready = true
 		conditions.MarkTrue(machineScope.LinodeMachine, clusterv1.ReadyCondition)
 	}
+
+	// Clean up after instance creation.
+	if linodeInstance.Status == linodego.InstanceRunning && machineScope.Machine.Status.Phase == "Running" {
```

Comment:
Do we need the second check here?

Reply:
Yeah, because the first status indicates of the state of the infrastructure and the second status is used to check if the node has successfully bootstrapped.